crates/proof-impl/evm-ee-stf/src/primitives.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::HashMap;
+
+use alloy_consensus::{serde_bincode_compat, Header};
+use reth_primitives::{
+    revm_primitives::alloy_primitives::{Address, Bytes, FixedBytes, B256},
+    TransactionSignedNoHash, Withdrawal,
+};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use strata_reth_primitives::WithdrawalIntent;
+use strata_state::block::ExecSegment;
+
+use crate::mpt::{MptNode, StorageEntry};
+
+/// Public Parameters that proof asserts
+pub type EvmEeProofOutput = Vec<ExecSegment>;
+
+/// Public Parameters that proof asserts
+pub type EvmEeProofInput = Vec<EvmBlockStfInput>;
+
+/// Result of the block execution
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct EvmBlockStfOutput {
+    pub block_idx: u64,
+    pub prev_blockhash: FixedBytes<32>,
+    pub new_blockhash: FixedBytes<32>,
+    pub new_state_root: FixedBytes<32>,
+    pub txn_root: FixedBytes<32>,
+    pub withdrawal_intents: Vec<WithdrawalIntent>,
+    pub deposits_txns_root: FixedBytes<32>,
```

Comment:
Would it make more sense to reorganize this so that fields that are computed are presented separately from fields that are merely re-published from the (private) input?  Also is `depeosits_txns_root` still being used?

Reply:
My bad, its should be array of deposits instead of `depeosits_txns_root`

Reply:
Public parameter from the EE is the `Vec<ExecSegment>`  which has separation as `UpdateInput` & `UpdateOutput`
https://github.com/alpenlabs/strata/blob/3d3bd74394301ad7cae2179d50fb6450b4c215ea/crates/proof-impl/evm-ee-stf/src/lib.rs#L111